modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/KeyValueModifyPlan.java

```diff
@@ -115,7 +115,7 @@ public IgniteKeyValueModify modifyNode() {
     @Override
     public <RowT> AsyncCursor<InternalSqlRow> execute(
             ExecutionContext<RowT> ctx,
-            @Nullable InternalTransaction tx,
+            InternalTransaction tx,
```

Comment:
seems additional assert will be useful ? 
assert tx.implicit();

Reply:
Key-value plan can be executed within an explicit (user-managed) transaction, in which case this assertion will fail  :thinking:

Reply:
I could put such assertion in `SelectCountPlan`, but I personally prefer to leave it as is since this plan does not require transaction at all.